Content.Shared/Placeable/PlaceableSurfaceSystem.cs

```diff
@@ -59,11 +64,24 @@ private void OnAfterInteractUsing(EntityUid uid, PlaceableSurfaceComponent surfa
         if (!_handsSystem.TryDrop(args.User, args.Used))
             return;
 
-        if (surface.PlaceCentered)
-            Transform(args.Used).LocalPosition = Transform(uid).LocalPosition + surface.PositionOffset;
-        else
-            Transform(args.Used).Coordinates = args.ClickLocation;
+        _transformSystem.SetCoordinates(args.Used,
+            surface.PlaceCentered ? Transform(uid).Coordinates.Offset(surface.PositionOffset) : args.ClickLocation);
 
         args.Handled = true;
     }
+
+    private void OnStorageInteractUsingAttempt(Entity<PlaceableSurfaceComponent> ent, ref StorageInteractUsingAttemptEvent args)
+    {
+        args.Cancelled = true;
+    }
+
+    private void OnStorageAfterOpen(Entity<PlaceableSurfaceComponent> ent, ref StorageAfterOpenEvent args)
+    {
+        SetPlaceable(ent.Owner, true, ent.Comp);
```

Comment:
make this method use Entity<> and check for the value before dirtying (if comp.value == paramValue) return ) while you are on this